NDTensors/ext/NDTEnsorsJLArraysExt/set_types.jl

```diff
@@ -0,0 +1,10 @@
+# TypeParameterAccessors definitions
+using NDTensors.TypeParameterAccessors: TypeParameterAccessors, Position
+using NDTensors.GPUArraysCoreExtensions: storagemode
+using JLArrays: JLArrays, JLArray
+
+function TypeParameterAccessors.default_type_parameters(::Type{<:JLArray})
+  return (Float64, 1)
+end
```

Comment:
We could consider making this a generic `AbstractArray` definition.

Reply:
This is something I was thinking about. I think one issue with that is wrappers don't always have ndims second for example 
```julia
julia> a = Matrix{Float32}(undef, (2,2))
julia> typeof(transpose(a))
LinearAlgebra.Transpose{Float32, Matrix{Float32}}
```

Reply:
Right, so then for those types we would require making a custom overload of `TypeParameterAccessors.default_type_parameters`, which would be required anyway.

Reply:
Should we make this default definition in this PR?

Reply:
You mean the default for all `AbstractArray` subtypes? Let's do it.

Reply:
I updated the code with a default `AbstractArray` version and call the default function for GPU tensors, except for Metal because it requires Float32.